sdks/python/apache_beam/runners/worker/operations.py

```diff
@@ -104,34 +104,45 @@ class Operation(object):
   one or more receiver operations that will take that as input.
   """
 
-  def __init__(self, operation_name, spec, counter_factory, state_sampler):
+  def __init__(self, name_context, spec, counter_factory, state_sampler):
     """Initializes a worker operation instance.
 
     Args:
-      operation_name: The system name assigned by the runner for this
-        operation.
+      name_context: A NameContext instance or string(deprecated), with the
+        name information for this operation.
       spec: A operation_specs.Worker* instance.
       counter_factory: The CounterFactory to use for our counters.
       state_sampler: The StateSampler for the current operation.
     """
-    self.operation_name = operation_name
+    if isinstance(name_context, common.NameContext):
+      #TODO(pabloem) - Clean this up once it's completely migrated.
```

Comment:
<!--new_thread; commit:fc000934fceaaeb8c8ca87017a50b4f2cbf35340; resolved:0-->
Space before TODO.  Can you also add a Jira for the remaining work?

Reply:
Done.